tests/test_eigen_tensor.cpp

```diff
@@ -5,12 +5,14 @@
     BSD-style license that can be found in the LICENSE file.
 */
 
-constexpr const char *test_eigen_tensor_module_name = "eigen_tensor";
-
 #define PYBIND11_TEST_EIGEN_TENSOR_NAMESPACE eigen_tensor
 
 #ifdef EIGEN_AVOID_STL_ARRAY
 #    undef EIGEN_AVOID_STL_ARRAY
 #endif
 
 #include "test_eigen_tensor.inl"
+
+#include "pybind11_tests.h"
+
+test_initializer name("eigen_tensor", test_module);
```

Comment:
I overlooked this before (in the old location): `name` is replaced by `eigen_tensor` in the `TEST_SUBMODULE` macro. It would be a little less surprising to use `eigen_tensor` here.

But, could this also be

```
test_initializer("eigen_tensor", test_module);
```

?

Reply:
That doesn't seem to compile. I just swapped it to 

`test_initializer eigen_tensor("eigen_tensor", test_module);`

as an alternative.

Reply:
Thanks for trying it out, and for the change!